test/dynamo/test_functions.py

```diff
@@ -2378,14 +2378,53 @@ def fn(inputs):
         input2 = torch.arange(2, dtype=torch.bfloat16)
         inputs = [input1, input2]
 
-        opt_fn = torch.compile()(fn)
+        opt_fn = torch.compile()(fn)  # will graph break
         self.assertEqual(opt_fn(inputs), fn(inputs))
 
         torch._dynamo.reset()
 
-        with self.assertRaises(torch._dynamo.exc.Unsupported):
-            opt_fn = torch.compile(fullgraph=True)(fn)
-            opt_fn(inputs)
+        opt_fn = torch.compile()(fn)  # will graph break
+        self.assertEqual(opt_fn(inputs), fn(inputs))
+
+    def test_filter_infinite_iterator(self):
+        def fn(x):
+            x = x + 1
+            return zip(range(3), filter(lambda x: x < 10, itertools.count()))
```

Comment:
Let's change the filter to `x > 10` to test `False` values as well? Also, input `x` is unused - let's at least return it (after applying `x = x + 1`).